src/vmm/src/devices/virtio/vhost_user_block/device.rs

```diff
@@ -303,3 +332,443 @@ impl VirtioDevice for VhostUserBlock {
+
+        impl VhostUserHandleBackend for MockMaster {
+            fn from_stream(sock: UnixStream, max_queue_num: u64) -> Self {
+                Self {
+                    sock,
+                    max_queue_num,
+                    is_owner: std::cell::UnsafeCell::new(false),
+                    features: 0,
+                    protocol_features: VhostUserProtocolFeatures::empty(),
+                    hdr_flags: std::cell::UnsafeCell::new(VhostUserHeaderFlag::empty()),
+                }
+            }
+
+            fn set_owner(&self) -> Result<(), vhost::Error> {
+                unsafe { *self.is_owner.get() = true };
+                Ok(())
+            }
+
+            fn set_hdr_flags(&self, flags: VhostUserHeaderFlag) {
+                unsafe { *self.hdr_flags.get() = flags };
+            }
+
+            fn get_features(&self) -> Result<u64, vhost::Error> {
+                Ok(self.features)
+            }
+
+            fn get_protocol_features(&mut self) -> Result<VhostUserProtocolFeatures, vhost::Error> {
+                Ok(self.protocol_features)
+            }
+
+            fn set_protocol_features(
+                &mut self,
+                features: VhostUserProtocolFeatures,
+            ) -> Result<(), vhost::Error> {
+                self.protocol_features = features;
+                Ok(())
+            }
+        }
+
+        let tmp_dir = utils::tempdir::TempDir::new().unwrap();
+        let tmp_dir_path_str = tmp_dir.as_path().to_str().unwrap();
+        let tmp_socket_path = format!("{tmp_dir_path_str}/tmp_socket");
+
+        unsafe {
+            let socketfd = libc::socket(libc::AF_UNIX, libc::SOCK_STREAM, 0);
+            if socketfd < 0 {
+                panic!("Cannot create socket");
+            }
+            let mut socket_addr = libc::sockaddr_un {
+                sun_family: libc::AF_UNIX as u16,
+                sun_path: [0; 108],
+            };
+
+            std::ptr::copy::<i8>(
+                tmp_socket_path.as_ptr().cast(),
+                socket_addr.sun_path.as_mut_ptr(),
+                tmp_socket_path.as_bytes().len(),
+            );
+
+            let bind = libc::bind(
+                socketfd,
+                (&socket_addr as *const libc::sockaddr_un).cast(),
+                std::mem::size_of::<libc::sockaddr_un>() as u32,
+            );
+            if bind < 0 {
+                panic!("Cannot bind socket");
+            }
+
+            let listen = libc::listen(socketfd, 1);
+            if listen < 0 {
+                panic!("Cannot listen on socket");
+            }
+        }
+
+        let vhost_block_config = VhostUserBlockConfig {
+            drive_id: "test_drive".to_string(),
+            partuuid: None,
+            is_root_device: false,
+            cache_type: CacheType::Unsafe,
+            socket: tmp_socket_path.clone(),
+        };
+        let vhost_block = VhostUserBlockImpl::<MockMaster>::new(vhost_block_config).unwrap();
+
+        assert_eq!(
```

Comment:
please add description of what the tests are verifying.

Reply:
added